fastlane_core/spec/project_spec.rb

```diff
@@ -296,12 +296,24 @@ def within_a_temp_dir
     end
 
     describe "build_settings() can handle empty lines" do
-      it "SUPPORTED_PLATFORMS should be iphonesimulator iphoneos" do
+      it "SUPPORTED_PLATFORMS should be iphonesimulator iphoneos on XCode >= 8.3" do
         options = { project: "./fastlane_core/spec/fixtures/projects/Example.xcodeproj" }
         @project = FastlaneCore::Project.new(options, xcodebuild_list_silent: true, xcodebuild_suppress_stderr: true)
-        expect(FastlaneCore::Project).to receive(:run_command).with("xcodebuild clean -showBuildSettings -project ./fastlane_core/spec/fixtures/projects/Example.xcodeproj 2> /dev/null", { timeout: 10, retries: 3, print: false }).and_return(File.read("./fastlane_core/spec/fixtures/projects/build_settings_with_toolchains"))
+        allow(FastlaneCore::Helper).to receive(:xcode_atleast?).and_return(true)
+        command = "xcodebuild -showBuildSettings -project ./fastlane_core/spec/fixtures/projects/Example.xcodeproj 2> /dev/null"
+        expect(FastlaneCore::Project).to receive(:run_command).with(command.to_s, { timeout: 10, retries: 3, print: false }).and_return(File.read("./fastlane_core/spec/fixtures/projects/build_settings_with_toolchains"))
         expect(@project.build_settings(key: "SUPPORTED_PLATFORMS")).to eq("iphonesimulator iphoneos")
       end
+      unless FastlaneCore::Helper.xcode_atleast?('8.3')
```

Comment:
the command is not really run in the tests.
`expect(FastlaneCore::Project).to receive(:run_command).with(command.to_s, { timeout: 10, retries: 3, print: false }).and_return(File.read("./fastlane_core/spec/fixtures/projects/build_settings_with_toolchains"))` mocks it.

the goal should be to test the following two scenarios.

## user has xcode < 8.3
  *  force `xcode_atleast?("8.3") ` to return `false`
  * expect the resulting command, to include `clean`

## user has xcode >= 8.3
  *  force `xcode_atleast?("8.3") ` to return `true`
  * expect the resulting command, to NOT include `clean`

by forcing the result we bypass the local installation interfering the unit test.